io/net/sriov_device_test.py

```diff
@@ -70,12 +72,49 @@ def setUp(self):
                                           default=None).split(' ')
         self.ipaddr = self.params.get('ipaddr', '*', default="").split(' ')
         self.netmask = self.params.get('netmasks', '*', default="").split(' ')
+        self.prefix = self.netmask_to_cidr(self.netmask)
         self.peer_ip = self.params.get('peer_ip', '*', default="").split(' ')
         self.mac_id = self.params.get('mac_id',
                                       default="02:03:03:03:03:01").split(' ')
         self.mac_id = [mac.replace(':', '') for mac in self.mac_id]
+        self.migratable = self.params.get('migratable', '*', default=0)
+        self.backup_device_type = self.params.get(
+            'backup_device_type', '*', default=None)
+        self.backup_device_slot_num = self.params.get(
+            'backup_device_slot_num', '*', default=None)
+        self.backup_veth_vnetwork = self.params.get(
+            'backup_veth_vnetwork', '*', default=None)
+        self.max_capacity = self.params.get('max_capacity', '*', default='10')
+        self.capacity = self.params.get('capacity', '*', default='2')
+        self.vios_name = self.params.get('vios_name', '*', default=None)
+        cmd = 'lssyscfg -m ' + self.server + \
+              ' -r lpar --filter lpar_names=' + self.vios_name + \
+              ' -F lpar_id'
+        self.vios_id = self.session.cmd(cmd).stdout_text.split()[0]
+        self.vnic_sriov_adapter = self.params.get(
+            'vnic_sriov_adapter', '*', default=None)
+        self.vnic_port_id = self.params.get('vnic_port_id', '*', default=None)
+        self.vnic_adapter_id = self.get_adapter_id(self.vnic_sriov_adapter)
+        self.priority = self.params.get('failover-priority', '*', default='50')
+        self.backup_vnic_backing_device = 'sriov/%s/%s/%s/%s/%s/%s/%s' % \
+        (self.vios_name, self.vios_id, self.vnic_adapter_id, self.vnic_port_id, \
+        self.capacity, self.priority, self.max_capacity)
         self.local = LocalHost()
 
+    @staticmethod
+    def netmask_to_cidr(netmask):
+        return(sum([bin(int(bits)).count("1") for bits in netmask.split(".")]))
+
+    @staticmethod
+    def get_adapter_id(slot):
+        cmd = "lshwres -m %s -r sriov --rsubtype adapter -F phys_loc:adapter_id" \
+              % (self.server)
+        output = self.session.cmd(cmd)
+        for line in output.stdout_text.splitlines():
+            if slot in line:
+                return line.split(':')[-1]
+        self.cancel("adapter not found at slot %s", slot)
```

Comment:
Is it fine to use self in a staticmethod ?

Reply:
looks like I havent pushed latest code